addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,16 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        notifications = []
+        for message in messages.sorted('author_id'):
+            partners = message.notification_ids.mapped('res_partner_id') | message.author_id
```

Comment:
Isn't this line why you have extra queries?
This is sending a notification to all recipients, instead of just the author and the current user.
It's not necessary to send it to recipients.